### ISSUE TITLE
Populated field type

### ISSUE THREAD

**Issue by wasya243** [typegoose/typegoose#811]
Hello! Please help me. Here is the code.

`import { prop, getModelForClass, Ref } from '@typegoose/typegoose';
import mongoose from 'mongoose';

export class Organization {
  @prop({required: true})
  name!: string;

  @prop({
    ref: () => User,
    foreignField: 'organizationId',
    localField: '_id'
  })
  public users: Ref<User>[] = [];
}
export class User {
  @prop({required: true})
  name!: string;

  @prop({
    ref: () => Organization, 
    required: true,
    foreignField: '_id',
    localField: 'organizationId',
    justOne: true
  })
  private organization!: Ref<Organization>;

  @prop({ required: true })
  public organizationId!: string;

}

const OrganizationModel = getModelForClass(Organization, {
  schemaOptions: {
    toJSON: { virtuals: true },
    toObject: { virtuals: true }
  }
});

const UserModel = getModelForClass(User, {
  schemaOptions: {
    toJSON: { virtuals: true },
    toObject: { virtuals: true }
  }
});
async function main() {
  try {
    await mongoose.connect('mongodb://localhost:27017/', {dbName: 'migration-to-typegoose'});

    const organization = await OrganizationModel.create({
      name: 'Acme Inc.'
    });

    const users = await Promise.all([
      UserModel.create({
        name: 'Vasyl',
        organizationId: organization._id.toString()
      }),
      UserModel.create({
        name: 'Misha',
        organizationId: organization._id.toString()
      })
    ]);

    const user = await UserModel.findOne().populate('organization');

    // here is TS error that name does not exist on organization
    console.log(user.organization.name)

    // possible solutions

    // 1) const user = await UserModel.findOne().populate<{organization: Organization}>('organization');
    // 2) same as above but define interface 
    // 3) typecasting to any
    // 5) if isDocument

    process.exit(0);
  } catch (err) {
    console.error(err);
    process.exit(1);
  }
}

main();
`
There are 2 entities: organization and user. Organization has many users. I do populate on user (populating organization). After that I try acccessing .name on populated organization field. I get TS error.

**Basically my question is there any other way to access fields on populated field without interfaces, typecasts, isDocument?**
